src/osd/ECBackend.cc

```diff
@@ -2456,6 +2513,44 @@ void ECBackend::objects_read_and_reconstruct(
   return;
 }
 
+bool ECBackend::should_partial_read(
+  const hobject_t &hoid,
+  std::list<ec_align_t> to_read,
+  const std::set<int> &want,
+  bool fast_read,
+  bool for_recovery)
+{
+    // Don't partial read if we are doing a fast_read
+    if (fast_read) {
+      return false;
+    }
+    // Don't partial read if the EC isn't systematic
+    if (!ec_impl->is_systematic()) {
+      return false;
+    }
+    // Don't partial read if we have multiple stripes
+    if (to_read.size() != 1) {
+      return false;
+    }
+    // Only partial read if the length is inside the stripe boundary
```

Comment:
These restrictions seem kind of brittle.  I don't see an obvious reason why the head and tail of a multi-stripe read couldn't be sub-stripe reads.  Shouldn't it simply be a function of the offsets?